vignettes/r-developer-guide.Rmd

```diff
@@ -24,129 +24,178 @@ knitr::read_chunk(path = system.file("plumber", "loess", "plumber.R", package =
                   labels = "loess")
 ```
 
-### Setup
-Install the R package.
+## Getting Started {#getting-started}
+
+This vignette will walk you through writing plumbertableau extensions in R. plumbertableau extensions are Plumber web services with a few additional pieces, and this vignette assumes some familiarity with Plumber (https://www.rplumber.io/). Plumber's [Quickstart guide](https://www.rplumber.io/articles/quickstart.html) gives a good overview of that package. Throughout this guide we'll use the terms "web service" and "API" (application programming interface) interchangeably.
+
+You can install the plumbertableau package from CRAN or GitHub.
 
-From CRAN:
 ```{r, eval = FALSE}
+# from CRAN
 install.packages("plumbertableau")
-```
 
-From Github:
-```{r, eval = FALSE}
+# from GitHub
 remotes::install_github("rstudio/plumbertableau")
 ```
 
-### Getting started {#getting-started}
-`plumbertableau`, as the name suggests, is an extension of the [`plumber`](https://www.rplumber.io/) package. If you are unfamiliar with `plumber`, the [quickstart guide](https://www.rplumber.io/articles/quickstart.html) provides a good overview of the package. Building a Tableau Analytics Extension with `plumbertableau` is simply a matter of building a `plumber` API with a few additional pieces.
+## What can plumbertableau Extensions Do? {#capabilities}
 
-### Creating an extension {#extension}
-Let's start with an example. Let's say we want to fit a loess curve to a given set of data. This can be done in R with the following:
-```{r}
+plumbertableau extensions respond to requests from Tableau. In a request, Tableau will provide an [*endpoint*](https://www.rplumber.io/articles/routing-and-input.html?q=endpoint#endpoints-1) — which roughly corresponds to an R function — and will send along one or more *arguments*, which will generally consist of data from Tableau.
+
+An extension *must* return data that's the same elements or rows as the arguments it receive. It can do things like transform text, or use a model to predict Y values from a matrix of X values. It can't, say, receive an integer representing the number of elements to sample, and return a vector of that length.
```

Comment:
```suggestion
An extension *must* return data that's the same number of observations as the arguments it receives. It can do things like transform text, or use a model to predict Y values from a matrix of X values. A plumbertableau extension can't, say, receive an integer representing the number of elements to sample, and return a vector of that length.
```